engine/flowCoef.cc

```diff
@@ -21,16 +21,46 @@
 #include "str.h"
 #include <ctype.h>
 #include <stdlib.h>
+#include <boost/spirit/include/qi.hpp>
+namespace qi = boost::spirit::qi;
 
 using namespace std;
+
+namespace
+// Extract leading number from a string. for ticket 1177. See https://stackoverflow.com/questions/45600212/regular-expression-multiple-floating-point.
+{
+  bool parseLine(std::string const& line, double& num) {
+      using It = std::string::const_iterator;
+      
+      // Construct grammar to skip in input formula
+      qi::rule<It, std::string()> nan = -qi::lit("1.0#") >> qi::no_case["nan"] >> -('(' >> *(qi::char_ - ')') >> ')');
+      qi::rule<It, std::string()> inf = qi::no_case[qi::lit("inf") >> -qi::lit("inity")];                                          
+      qi::rule<It, std::string()> sign =   qi::lit('+') | '-';
+      qi::rule<It, std::string()> skip = qi::char_( '/' ) | qi::char_( '\\' ) | qi::char_( '_' );
+  
```

Comment:
Yikes!!!

Reply:
I think maybe your idea of just testing for nan or inf at start, maybe the way to go. Alternatively, test for digit or decimal point as the first character and run strtod, otherwise just treat as string.